apps/backend/src/modules/plan-pdf-export/plan-pdf-export.service.ts

```diff
@@ -110,14 +88,98 @@ class PlanPdfExportService {
 			width,
 		});
 
-		await page.goto(url.toString(), { waitUntil: NETWORK_IDLE_0 });
+		await page.goto(url.toString(), {
+			waitUntil: PLAN_PDF_EXPORT_NETWORK_IDLES.IDLE_0,
+		});
 
 		const pngBuffer = await page.screenshot({ fullPage: true, type: "png" });
 
 		await browser.close();
 
 		return Buffer.from(pngBuffer as Buffer);
 	}
+
+	private buildPrintUrl(dto: ExportPlanPdfDto): string {
+		const baseUrl =
+			config.ENV.FRONTEND.PLAN_PRINT_URL || PLAN_PDF_EXPORT_ROUTES.ROOT;
+		const hasPath = baseUrl.includes(PLAN_PDF_EXPORT_ROUTES.PLAN_STYLE_PRINT);
+		const path = hasPath ? "" : PLAN_PDF_EXPORT_ROUTES.PLAN_STYLE_PRINT;
+
+		return dto.planStyle
+			? `${baseUrl}${path}?style=${encodeURIComponent(dto.planStyle)}&planId=${dto.planId?.toString() ?? ""}`
+			: `${baseUrl}${path}?planId=${dto.planId?.toString() ?? ""}`;
+	}
+
+	private async cleanupBrowser(browser: puppeteer.Browser): Promise<void> {
+		await browser.close();
+	}
+
+	private async generatePdfBuffer(
+		page: puppeteer.Page,
+		dto: ExportPlanPdfDto,
+	): Promise<Buffer> {
+		const pdfBuffer = await page.pdf({
+			format: dto.format,
+			preferCSSPageSize: true,
+			printBackground: true,
+		});
+
+		return Buffer.from(pdfBuffer);
+	}
+
+	private async launchBrowser(): Promise<puppeteer.Browser> {
+		return await puppeteer.launch();
```

Comment:
```suggestion
		return await puppeteer.launch({
		    headless: "new",
		    args: [
		        "--no-sandbox",
		        "--disable-setuid-sandbox"
		    ]
		});
```
Let's try it in this way to make it work on deploy. Please, let me know if this fix does not work locally

Reply:
It works, but the problem is that there is no "new" option. Only:
headless: true which launches the browser in a new headless mode
headless: "shell" which launches old headless mode (shell)

Which one of those will be better for you here?

Reply:
let's go with true